src/storage/v2/vertex_info_helpers.hpp

```diff
@@ -0,0 +1,215 @@
+
+#include <algorithm>
+#include <tuple>
+#include <vector>
+namespace memgraph::storage {
+
+template <Delta::Action>
+struct DeltaAction_tag {};
+
+template <Delta::Action A, typename Method>
+struct ActionMethodImpl : Method {
+  // Uses tag dispatch to ensure method is only called for the correct action
+  void operator()(DeltaAction_tag<A> /*unused*/, Delta const &delta) { Method::operator()(delta); }
+};
+
+template <Delta::Action A, typename Method>
+inline auto ActionMethod(Method &&func) {
+  return ActionMethodImpl<A, Method>{std::forward<Method>(func)};
+}
+
+/// Converts runtime Delta::Action into compile time tag, this allows us to dispatch to the correct overload
+template <typename Func>
+inline void DeltaDispatch(Delta const &delta, Func &&func) {
+  // clang-format off
+#define dispatch(E) case E: return func(DeltaAction_tag<E>{}, delta); // NOLINT
+  // clang-format on
+  switch (delta.action) {
+    using enum Delta::Action;
+    dispatch(DELETE_DESERIALIZED_OBJECT);
+    dispatch(DELETE_OBJECT);
+    dispatch(RECREATE_OBJECT);
+    dispatch(SET_PROPERTY);
+    dispatch(ADD_LABEL);
+    dispatch(REMOVE_LABEL);
+    dispatch(ADD_IN_EDGE);
+    dispatch(ADD_OUT_EDGE);
+    dispatch(REMOVE_IN_EDGE);
+    dispatch(REMOVE_OUT_EDGE);
+  }
+#undef dispatch
+}
+
+inline auto Exists_ActionMethod(bool &exists) {
+  using enum Delta::Action;
+  // clang-format off
+  return utils::Overloaded{
+      ActionMethod<DELETE_DESERIALIZED_OBJECT>([&](Delta const & /*unused*/) { exists = false; }),
+      ActionMethod<DELETE_OBJECT>([&](Delta const & /*unused*/) { exists = false; })
+  };
+  // clang-format on
+}
+
+inline auto Deleted_ActionMethod(bool &deleted) {
+  using enum Delta::Action;
+  return ActionMethod<RECREATE_OBJECT>([&](Delta const & /*unused*/) { deleted = false; });
+}
+
+inline auto HasLabel_ActionMethod(bool &has_label, LabelId label) {
+  using enum Delta::Action;
+  // clang-format off
+  return utils::Overloaded{
+      ActionMethod<REMOVE_LABEL>([&, label](Delta const &delta) {
+        if (delta.label == label) {
+          MG_ASSERT(has_label, "Invalid database state!");
+          has_label = false;
+        }
+      }),
+      ActionMethod<ADD_LABEL>([&, label](Delta const &delta) {
+        if (delta.label == label) {
+          MG_ASSERT(!has_label, "Invalid database state!");
+          has_label = true;
+        }
+      })
+  };
+  // clang-format on
+}
+
+inline auto Labels_ActionMethod(std::vector<LabelId> &labels) {
+  using enum Delta::Action;
+  // clang-format off
+  return utils::Overloaded{
+      ActionMethod<REMOVE_LABEL>([&](Delta const &delta) {
+        auto it = std::find(labels.begin(), labels.end(), delta.label);
+        MG_ASSERT(it != labels.end(), "Invalid database state!");
+        *it = labels.back();
+        labels.pop_back();
+      }),
+      ActionMethod<ADD_LABEL>([&](Delta const &delta) {
+        auto it = std::find(labels.begin(), labels.end(), delta.label);
```

Comment:
Could we add std::execution::unseq to these std::finds?

Reply:
I did try that, need to make sure its removed. The expected number of labels is low which means cost of SIMD here makes perf worse.

Reply:
Oh so this was an interesting insight. I think the code gen for std::find std::execution::unseq is a poor choice in this situation. Its not SIMD, but lots of differences that probably also make it not ideal. https://godbolt.org/z/M58bxeqcM.

Reply:
Cool! Yeah, I am not sure how mature `std::execution::unseq` actually is. 
Changing the compile to clang 15 seems to do a good job.